main.py

```diff
@@ -311,57 +291,84 @@ def on_draw(self, widget, cr):
         cr.paint()
 
         # Draw the visualization
-        #this will redraw the whole screen even though very little changes between frames causeing low end gpus to melt.
-        #it will also just stop drawing sometimes on lower end gpus. 
         if hasattr(self, 'sample'):
+            #all of this stuff may be able to be calculated in advance.
```

Comment:
on the next line I say why I cant do that.
if the window is resized it will still do calculations as if it was not.
also I'm going to do a full code refactor where I will split up tasks into multiple files so ill do it after that.
ps. I dont know if you code tells you this but I am the sole maintainer of this repo.